grid/src/Cabana_Grid_Splines.hpp

```diff
@@ -451,6 +451,586 @@ struct Spline<3>
+                     xn * ( 2220.0 +
+                            xn * ( -1600.0 +
+                                   xn * ( 240.0 +
+                                          xn * ( 384.0 +
+                                                 xn * ( -192.0 ) ) ) ) ) ) ) *
+            denom_2;
+
+        // Knot at i - 1
+        values[2] =
+            ( 10543.0 +
+              xn * ( -17340.0 +
+                     xn * ( 4740.0 +
+                            xn * ( 6880.0 +
+                                   xn * ( -4080.0 +
+                                          xn * ( -960.0 +
+                                                 xn * ( 960.0 ) ) ) ) ) ) ) *
+            denom_31;
+
+        // Knot at i
+        values[3] =
+            ( 5887.0 +
+              xn * xn *
+                  ( -4620.0 + xn * xn * ( 1680.0 + xn * xn * ( -320.0 ) ) ) ) *
+            denom_0;
+
+        // Knot at i + 1
+        values[4] =
+            ( 10543.0 +
+              xn * ( 17340.0 +
+                     xn * ( 4740.0 +
+                            xn * ( -6880.0 +
+                                   xn * ( -4080.0 +
+                                          xn * ( 960.0 +
+                                                 xn * ( 960.0 ) ) ) ) ) ) ) *
+            denom_31;
+
+        // Knot at i + 2
+        values[5] =
+            ( 361.0 +
+              xn * ( 1416.0 +
+                     xn * ( 2220.0 +
+                            xn * ( 1600.0 +
+                                   xn * ( 240.0 +
+                                          xn * ( -384.0 +
+                                                 xn * ( -192.0 ) ) ) ) ) ) ) *
+            denom_2;
+
+        // Knot at i + 3
+        values[6] =
+            ( 1.0 +
+              xn * ( 12.0 +
+                     xn * ( 60.0 +
+                            xn * ( 160.0 +
+                                   xn * ( 240.0 +
+                                          xn * ( 192.0 +
+                                                 xn * ( 64.0 ) ) ) ) ) ) ) *
+            denom_31;
+    }
+
+    /*!
+      \brief Calculate the value of the gradient of the spline in the physical
+      frame.
+      \param x0 The coordinate at which to evaluate the spline in the logical
+      grid space.
+      \param rdx The inverse of the physical distance between grid locations.
+      \param gradients Basis gradient values at the knots in the physical frame.
+      Ordered from lowest to highest in terms of knot location.
+    */
+    template <typename Scalar>
+    KOKKOS_INLINE_FUNCTION static void
+    gradient( const Scalar x0, const Scalar rdx, Scalar gradients[num_knot] )
+    {
+        // Constants
+        Scalar denom_3 = 1.0 / 3840.0;
+        Scalar denom_2 = denom_3 * 4.0;
+        Scalar denom_1 = denom_3 * 5.0;
+        Scalar denom_0 = denom_3 * 40.0;
+
+        // Knot at i - 2
+        Scalar xn = x0 - static_cast<int>( x0 );
+        xn -= 0.5;
```

Comment:
Same as for quartic values. Precompute the powers of `xn` and change formatting to improve readability of the calls for `values[n]=`